backend/search/basex_search.py

```diff
@@ -0,0 +1,74 @@
+class DatabaseSearcher:
```

Comment:
I'm not sure I like the design of this class. Could you add an example of how you intend to use it?

If you'd like the class to retain state, maybe call it something like `BaseXQuery`. To me, `DatabaseSearcher` implies something more generic, but currently an object is tied to a specific db and a specific query.
Otherwise, you could maybe make every method here static, and not keep any internal state.